packages/lib/gpc/src/gpcCompile.ts

```diff
@@ -132,13 +132,12 @@ function prepCompilerMaps<
 
     // If the object input is undefined, e.g. if nothing is revealed, add the
     // object name to the map anyway for later reference.
-    objMap.set(
-      objName,
-      objInput === undefined ? undefined : { objConfig, objInput, objIndex }
-    );
+    objMap.set(objName, { objConfig, objInput, objIndex });
```

Comment:
Why does this not require changing the ObjInput type to include undefined?  I would've expected that to be necessary since it's now going to sometimes be undefined.

I think it's worth explicitly marking that objInput can be undefined in CompilerObjInfo and CompilerEntryInfo types so that use sites can be checked.